File: src/main/java/axirassa/ioc/HibernateTestingModule.java

package axirassa.ioc;

import org.apache.tapestry5.hibernate.HibernateConfigurer;
import org.apache.tapestry5.hibernate.HibernateCoreModule;
import org.apache.tapestry5.ioc.ServiceBinder;
import org.apache.tapestry5.ioc.annotations.SubModule;
import org.hibernate.cfg.Configuration;

import axirassa.util.HibernateTools;
import axirassa.util.test.HibernateCleanupService;

@SubModule(HibernateCoreModule.class)
public class HibernateTestingModule {
	public static void bind(ServiceBinder binder) {
		binder.bind(HibernateCleanupService.class);
	}


	public static HibernateConfigurer decorateDefaultHibernateConfigurer(HibernateConfigurer configuration) {
		return new HibernateTestConfigurer();
	}
}

class HibernateTestConfigurer implements HibernateConfigurer {
	@Override
	public void configure(Configuration configuration) {
		configuration.configure();
		HibernateTools.configureTesting(configuration);
	}
}

File: src/main/java/axirassa/ioc/HibernateTestingModule.scala

package axirassa.ioc

import org.apache.tapestry5.ioc.MappedConfiguration
import org.hibernate.Session

import axirassa.util.HibernateTools

class HibernateTestingModule {
	var database : Session = _


	private def getHibernateSession () = {
		if (database == null)
			database = HibernateTools.buildTestingSessionFactory().openSession()

		database
	}


	def contributeServiceOverride (configuration :MappedConfiguration[Class[_], Object] ) {
		configuration.add(classOf[Session], getHibernateSession())
	}
}
